Sources/EdgeNetworkHandlers/ResponseCallbackHandler.swift

```diff
@@ -39,32 +51,93 @@ class ResponseCallbackHandler {
+        if let responseHandler = responseHandlers[requestEventId] {
+            responseHandler.onComplete()
             responseHandlers[requestEventId] = nil
-            Log.trace(label: TAG, "Removing callback for Edge response with request unique id \(requestEventId).")
         }
+
+        if let completionHandler = completionHandlers[requestEventId] {
+            completionHandler(edgeEventHandles[requestEventId] ?? [], edgeEventErrors[requestEventId] ?? [])
+            completionHandlers[requestEventId] = nil
+        }
+
+        edgeEventHandles[requestEventId] = nil
+        edgeEventErrors[requestEventId] = nil
+
+        Log.trace(label: TAG, "Removing completion handlers for Edge response with request unique id \(requestEventId).")
+    }
+
+    /// Updates the list of `EdgeEventHandle`(s) for current `requestEventId` and calls onResponseUpdate if a `ResponseHandler` is registered.
+    /// - Parameters:
+    ///   - eventHandle: newly received event handle
+    ///   - requestEventId: the request event identifier associated with this event handle
+    func eventHandleReceived(_ eventHandle: EdgeEventHandle, requestEventId: String?) {
+        guard let unwrappedRequestEventId = requestEventId, !unwrappedRequestEventId.isEmpty else { return }
+        if edgeEventHandles[unwrappedRequestEventId] != nil {
+            edgeEventHandles[unwrappedRequestEventId]?.append(eventHandle)
+        } else {
+            edgeEventHandles[unwrappedRequestEventId] = [eventHandle]
+        }
+        invokeResponseHandler(eventHandle: eventHandle, eventError: nil, requestEventId: unwrappedRequestEventId)
     }
 
-    /// Invokes the response handler for the unique event identifier (if any callback was previously registered for this id).
-    /// - Parameter eventData: data received from the Edge response event, containing the event handle payload and the request event identifier
-    /// - Parameter requestEventId: the request event identifier to be called with the provided `eventData`
-    func invokeResponseHandler(eventData: [String: Any], requestEventId: String?) {
+    /// Updates the list of `EdgeEventError`(s) for current `requestEventId` and calls onErrorUpdate if a `ResponseHandler` is registered.
+    /// - Parameters:
+    ///   - eventError: newly received error/warning message
+    ///   - requestEventId: the request event identifier associated with this event handle
+    func eventErrorReceived(_ eventError: EdgeEventError, requestEventId: String?) {
         guard let unwrappedRequestEventId = requestEventId, !unwrappedRequestEventId.isEmpty else {
             return
         }
+        if edgeEventErrors[unwrappedRequestEventId] != nil {
+            edgeEventErrors[unwrappedRequestEventId]?.append(eventError)
+        } else {
+            edgeEventErrors[unwrappedRequestEventId] = [eventError]
+        }
+        invokeResponseHandler(eventHandle: nil, eventError: eventError, requestEventId: unwrappedRequestEventId)
```

Comment:
Same comment as about regarding only calling `invokeResponseHandler` if the requestEventId is in the responseHandlers' dictionary.